src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java

```diff
@@ -44,34 +50,44 @@ public class TransactionSendClosureHandler implements CommandHandler<Transaction
     @Autowired
     private TransactionsEventStoreRepository<Object> eventStoreRepository;
 
+    @Autowired
+    NotificationsServiceClient notificationsServiceClient;
+
     @Override
     public Mono<TransactionClosureSentEvent> handle(TransactionClosureSendCommand command) {
-        TransactionActivated transaction = command.getData().transaction();
-
-        if (transaction.getStatus() != TransactionStatusDto.AUTHORIZED) {
-            log.error("Error: requesting closure status update for transaction in state {}", transaction.getStatus());
-            return Mono.error(new AlreadyProcessedException(transaction.getRptId()));
-        } else {
-            UpdateAuthorizationRequestDto updateAuthorizationRequest = command.getData().updateAuthorizationRequest();
-            return authorizationRequestedEventStoreRepository.findByTransactionIdAndEventCode(
-                            transaction.getTransactionId().value().toString(),
-                            TransactionEventCode.TRANSACTION_AUTHORIZATION_REQUESTED_EVENT
-                    )
-                    .switchIfEmpty(Mono.error(new TransactionNotFoundException(transaction.getTransactionActivatedData().getPaymentToken())))
-                    .flatMap(authorizationRequestedEvent -> {
-                        TransactionAuthorizationRequestData authorizationRequestData = authorizationRequestedEvent.getData();
+        Mono<Transaction> transaction = replayTransactionEvents(command.getData().transaction().getTransactionId().value());
+
+        Mono<? extends BaseTransaction> alreadyProcessedError = transaction
+                .cast(BaseTransaction.class)
+                .doOnNext(t -> log.error("Error: requesting closure status update for transaction in state {}", t.getStatus()))
+                .flatMap(t -> Mono.error(new AlreadyProcessedException(t.getRptId())));
+
+        return transaction
+                .cast(BaseTransaction.class)
+                .filter(t -> t.getStatus() == TransactionStatusDto.AUTHORIZED)
+                .switchIfEmpty(alreadyProcessedError)
+                .cast(TransactionWithCompletedAuthorization.class)
+                .flatMap(tx -> {
+                    UpdateAuthorizationRequestDto updateAuthorizationRequest = command.getData().updateAuthorizationRequest();
+                    return authorizationRequestedEventStoreRepository.findByTransactionIdAndEventCode(
+                                    tx.getTransactionId().value().toString(),
+                                    TransactionEventCode.TRANSACTION_AUTHORIZATION_REQUESTED_EVENT
```

Comment:
This check could be redundant since in the previous flatMap the transaction is filtered by its status `.filter(t -> t.getStatus() == TransactionStatusDto.AUTHORIZED)`